server/index.js

```diff
@@ -29,43 +22,22 @@ app.get('*.js', function(req, res, next) {
   }
   next();
 });
-
 app.use(express.static(__dirname + '/../public'));
 app.use('/:id', express.static(__dirname + '/../public'));
 
 
+// handle endpoints
+
 //get one item's description
-app.get('/description/:productId', (req, res, next) => {
-  let productId = req.params.productId;
-  query.getDescriptionForOneProduct(productId, (err, description) => {
-    if (description.length === 0) {
-      res.sendStatus(404);
-      next();
-    } else {
-      let itemData = description;
-      res.status(200).send(itemData);
-      next();
-    }
-  });
-});
+app.route('/description/:productId')
+  .get(descriptionById)
+  .put(genUpdate, updateDescription)
+  .delete(deleteDescription);
```

Comment:
This is CLEAN! I've never seen it done this way before. Your controllers file along with this really makes for excellent code clarity.

Reply:
express has some really nifty features like this for cleaning up routers.